### PR TITLE
feat: end to end functionality for registering ui to servers and on demand scaling

### PR DESCRIPTION
**Pull request by syllabix**

### CODE REVIEW

**Comment by syllabix** [comment]
@muratozkan and @bhavaniMiro - this is almost ready to merge. 
I would like to always maintain that local no docker / no k8s environments work which will require some small modifications to the new `api` service as well as updates to the docs.

File: server/src/handler/board.rs [hunk 1]
```diff
@@ -73,12 +69,13 @@ fn get_space_id(req: &HttpRequest) -> Result<space::ID, Error> {
 pub async fn connect(
     req: HttpRequest,
     stream: web::Payload,
+    token: web::Query<Token>,
     spaces: web::Data<board::Registry>,
-    users: web::Data<user::Registry>,
+    user_client: web::Data<user::Client>,
 ) -> Result<HttpResponse, Error> {
-    let user = get_user(&req, users)?;
+    let user = get_user(token.tk, user_client).await?;
```
Comment (bhavaniMiro) [inline server/src/handler/board.rs:76]: QQ: We still expect that the token encodes the user ID in plain right? (well, at least for now). Do you still imgaine it possible the token could technically come from a cookie? Please correct me if I'm wrong :) Just trying to reason about `token.tk` here but the `user_id` signature ion line 45

Comment (syllabix) [reply server/src/handler/board.rs:76]: i'd like to move away from this. 
it really is bad practice and pretty much just some quick duct tape for the demo. 
we should have a proper signed token that can at a minimum map to a user (ideally a session). 
that said, we cannot reliably trust cookies in websocket upgrade requests. 
ideally we should move to this to some kind of "first ping is an auth challenge" - 
In this case - I moved it to a query string param.

**Comment by syllabix** [comment]
@muratozkan and @bhavaniMiro - we should be good to go!

**Review by muratozkan — APPROVED** [review]
looks good to me - there'll be some changes coming up in #15 about the SDK usage

**Pull request merged**